code/modules/mob/logout.dm

```diff
@@ -2,6 +2,8 @@
 	SStgui.on_logout(src)
 	unset_machine()
 	player_list -= src
+	if(ckey in current_watchlist)

```

Comment:
You don't need this check, you can just try to remove them from the list(you don't need to know if they're in it)